src/cocotb/_scheduler.py

```diff
@@ -589,24 +522,31 @@ def _resume_task_upon(self, task, trigger):
                 raise InternalError("More than one task waiting on an unprimed trigger")
 
             try:
-                trigger._prime(self._react)
+                if isinstance(trigger, GPITrigger):
+                    trigger._prime(self._gpi_react)
+                else:
+                    trigger._prime(self._react)
```

Comment:
I'm thinking I'll just go back to the one `_react` call in which I'll put in a `if isinstance(trigger, GPITrigger)` guard for the GPITrigger specific stuff.

Reply:
Ehhh, maybe something different. Since we are using context managers for profiling `_gpi_react` wrapping `_react` is useful.